File: Back-End/app/websockets/managers/user_manager.py
from app.websockets.models.user import User
from app.constants import ClassroomUserRole
from fastapi import WebSocket
from app.websockets.models.classroom import Classroom


class UserManager:
    def create_student(self, user_id, classroom: Classroom, websocket: WebSocket):
        new_student = User(user_id=user_id,
                           role=ClassroomUserRole.STUDENT, websocket=websocket)
        for assignment in classroom.assignments:
            new_student.add_user_assignment(
                assignment.to_user_assignment(user_id=user_id))
        return new_student

    def create_teacher(self, user_id, websocket: WebSocket):
        new_teacher = User(user_id=user_id,
                           role=ClassroomUserRole.TEACHER, websocket=websocket)
        return new_teacher


user_manager = UserManager()

File: Back-End/rust/src/managers/user_manager.rs
use warp::ws::WebSocket;

use crate::models::{classroom::Classroom, user::User};

pub struct UserManager;

impl UserManager {
    pub fn new() -> UserManager {
        todo!()
    }

    pub fn create_student(
        &mut self,
        user_id: &str,
        classroom: &Classroom,
        websocket: WebSocket,
    ) -> User {
        todo!()
    }

    pub fn create_teacher(&mut self, user_id: &str, websocket: WebSocket) -> User {
        todo!()
    }
}
